task/build-vm-image/0.1/build-vm-image.yaml

```diff
@@ -155,19 +155,19 @@ spec:
         ssh -v $SSH_ARGS "$SSH_HOST"  mkdir -p "$BUILD_DIR/workspaces" "$BUILD_DIR/scripts" "$BUILD_DIR/tmp" "$BUILD_DIR/tekton-results" "$BUILD_DIR/entitlement"
 
         if [ ! -n "${CONFIG_TOML_FILE}" ]; then
-          echo "No CONFIG_TOML_FILE specified"
+          echo "No CONFIG_TOML_FILE specified. Assuming config.toml"
           export CONFIG_TOML_FILE=config.toml
-          if [ -f /var/workdir/source/config.toml ]; then
-            echo "Using the config.toml file found in the repository root!"
-            echo "  Remove the config.toml file or set params.CONFIG_TOML_FILE to another file to prevent using config.toml."
-          else
-            echo "No config.toml file found. Using an empty configuration."
-            touch /var/workdir/source/$CONFIG_TOML_FILE
-          fi
         fi
-        echo "Using the following config.toml file $CONFIG_TOML_FILE:"
-        cat /var/workdir/source/$CONFIG_TOML_FILE
 
+        if [ -f "/var/workdir/source/${CONFIG_TOML_FILE}" ]; then
+          echo "Using the ${CONFIG_TOML_FILE} file found in the repository."
+        else
+          echo "No ${CONFIG_TOML_FILE} file found. Using an empty configuration."
+          touch "/var/workdir/source/${CONFIG_TOML_FILE}"
+        fi
```

Comment:
With the current version of the task, explicitly passing a non-existent CONFIG_TOML_FILE would result in a failure (intentionally) rather than ignoring the problem and creating an empty config.

IMO that's the saner behavior that one would expect from most tools, but there seems to be a natural pull towards creating the empty config instead (both in your and in Andrew's PRs :smile: )

I'm fine with it either way, the log message should make it clear why e.g. a mis-spelled config file path is being ignored

Reply:
Aha. Now that you say that, I do remember those iterations two weeks back. I think that the reason that both @ralphbean and myself did that is because it enables cleaner/clearer logic. 

Once I tried to allow failures when a nonexistent file is specified, the logic became more confusing.

Reply:
See the table for intended behavior from the prior PR: https://github.com/konflux-ci/build-definitions/pull/1171#discussion_r1686588637

Reply:
I wouldn't say it _enables_ cleaner logic, it's just a touch easier to make it clean. Can still be clean and behave nicely though

```bash
if [ -z "$CONFIG_TOML_FILE" ]; then
    CONFIG_TOML_FILE=config.toml
    touch "$CONFIG_TOML_FILE"
elif ! [ -f "$CONFIG_TOML_FILE" ]; then
    echo "Config file doesn't exist: $CONFIG_TOML_FILE"
    exit 1
fi
```

Reply:
I fear I've made a mess. I wasn't aware of the conversations you guys had the other week about this and I assumed this section of code was just carelessly written in the first place during fast-hacking to make vm building work in early June, when in reality you guys had really thought it through. The code was written more recently than I knew!

My real goal here was to get https://gitlab.com/redhat/rhel-ai/disk-images/nvidia-bootc/-/commit/3ab9e272069e9469abc2ace36bf11b63a438f456 to happen - and it did.

I'm inclined to _drop_ this PR without merging it, just to let it exist as you guys defined it in https://github.com/konflux-ci/build-definitions/pull/1171#discussion_r1686588637. WDYT?

Reply:
:+1:

Reply:
@ralphbean , @chmeliik 's suggestion above is similar to what you proposed. It still defines the failure criteria and clearly describes why a failure happens instead of falling back to other processes to fail (i.e. rsync). I think this would still be an improvement.

Reply:
Ack, if we adopt that - we'll need a version bump per [this comment](https://github.com/konflux-ci/build-definitions/pull/1222#discussion_r1697548103), true?

Reply:
No, If the only failure is a user specifying a file path that does not exist then that is consistent. The breaking change previously is if no file was specified. We would not create an empty file for them.